lambda-utils/lib/resolved-promise-is-success.ts

```diff
@@ -0,0 +1,20 @@
+import middy from "@middy/core";
+import {APIGatewayProxyEventAnyVersion, APIGatewayProxyResultAnyVersion} from "./types";
+
+/**
+ * Middleware to allow an async handler to return its exact response body.
+ * This middleware will wrap it up as an APIGatewayProxyResult.
+ * Must be registered as the last (thus first to run) "after" middleware.
+ */
+export const resolvedPromiseIsSuccessMiddleware = (): middy.MiddlewareObj<APIGatewayProxyEventAnyVersion, APIGatewayProxyResultAnyVersion> => ({
+    after: async (request) => {
+        // If response isn't a proper API result object, convert it into one.
+        let response = request.response;
+        if (!response || typeof response !== 'object' || (!response.statusCode && !response.body)) {
+            request.response = {
+                statusCode: 200,
```

Comment:
If status code was provided, but not body, it will not be honored, but replaced by this value.  Similar issue with body.

Reply:
If status is provided, then response will be an object and the condition on line 13 will fail.

The use case here if for a lambda handler that successfully returns an object, and the middleware handles serializing the body and setting the status code. If you don't want this middleware to kick in, the user needs to minimally provide the normal `APIGatewayProxyResult` with `statusCode` and `body`.

I can see your argument that someone may want to provide just one (isn't statusCode required?), but then this middleware would fail to do its job if the response object happens to have a `body` property - so it can go wrong either way. This preserves existing behavior going back to v1.0 of this library. It also makes a V1 handler function very much like the new V2 handler, with the exception that you have the option to bypass it by returning _both_ statusCode and body.